open-models/use-cases/guess_app.ipynb

```diff
@@ -0,0 +1,816 @@
+      },
+      "outputs": [],
+      "source": [
+        "import os\n",
+        "\n",
+        "from google.cloud import aiplatform\n",
+        "\n",
+        "PROJECT_ID = \"[your-project-id]\"  # @param {type:\"string\", isTemplate: true}\n",
+        "if PROJECT_ID == \"[your-project-id]\":\n",
+        "    PROJECT_ID = str(os.environ.get(\"GOOGLE_CLOUD_PROJECT\"))\n",
+        "\n",
+        "LOCATION = os.environ.get(\"GOOGLE_CLOUD_REGION\", \"us-central1\")\n",
+        "\n",
+        "aiplatform.init(project=PROJECT_ID, location=LOCATION)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ee37e1544281"
+      },
+      "source": [
+        "### Requirements"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "877cd3fb2dce"
+      },
+      "source": [
+        "To successfully deploy your Hugging Face model on Vertex AI, you need to have the following IAM roles set:\n",
+        "\n",
+        "- Artifact Registry Reader (roles/artifactregistry.reader)\n",
+        "- Vertex AI User (roles/aiplatform.user)\n",
+        "\n",
+        "For more information about granting roles, see [Manage access](https://cloud.google.com/iam/docs/granting-changing-revoking-access).\n",
+        "\n",
+        "---\n",
+        "\n",
+        "You also need to enable the following APIs (if not enabled already):\n",
+        "\n",
+        "- Vertex AI API (aiplatform.googleapis.com)\n",
+        "- Artifact Registry API (artifactregistry.googleapis.com)\n",
+        "\n",
+        "For more information about API enablement, see [Enabling APIs](https://cloud.google.com/apis/docs/getting-started#enabling_apis).\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "2pNoqLHVWG6X"
+      },
+      "source": [
+        "### Import libraries"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "V41joDcRWIzY"
+      },
+      "outputs": [],
+      "source": [
+        "import base64\n",
+        "import io\n",
+        "import os\n",
+        "\n",
+        "from IPython.display import display\n",
+        "from PIL import Image\n",
+        "from google.cloud.aiplatform import Endpoint\n",
+        "import gradio as gr\n",
+        "from huggingface_hub import get_token\n",
+        "import vertexai\n",
+        "from vertexai.generative_models import GenerativeModel, SafetySetting"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "2qxTWPNu1p-d"
+      },
+      "source": [
+        "# Build a \"Guess what or who\" app\n",
+        "\n",
+        "To deploy a \"Guess what or who\" application using a Hugging Face model like FLUX for image generation on Vertex AI, you first register the chosen model within the Vertex AI Model Registry.  This registry serves as a central repository for managing your models on Vertex AI.\n",
```

Comment:
Add a link to the FLUX model and Vertex AI Model Registry docs

Reply:
done